src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/ExcelAdaptorServiceImpl.java

```diff
@@ -32,16 +33,25 @@
 import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.FILE_NO_DATA_ERROR_MESSAGE;
 import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.FILE_NO_VALID_SHEET_ERROR_MESSAGE;
 import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.IS_PRIMARY_FIELD;
+import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.REQUIRED_CW_SHEET_NAME;
+import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.REQUIRED_ROLE_MAPPING_SHEET_NAME;
 
 @Service
 @SuppressWarnings("unchecked")
 public class ExcelAdaptorServiceImpl implements ExcelAdaptorService {
+    private FormulaEvaluator formulaEvaluator = null;
 
     public <T> List<T> parseExcel(Workbook workbook, Class<T> classType) {
+        formulaEvaluator = workbook.getCreationHelper().createFormulaEvaluator();
         if (workbook.getNumberOfSheets() < 1) { // check at least 1 sheet present
             throw new ExcelValidationException(HttpStatus.BAD_REQUEST, FILE_NO_DATA_ERROR_MESSAGE);
         }
-        Sheet sheet = workbook.getSheet(CaseWorkerConstants.REQUIRED_SHEET_NAME);
+        Sheet sheet;
+        if (classType.isAssignableFrom(CaseWorkerProfile.class)) {
+            sheet = workbook.getSheet(REQUIRED_CW_SHEET_NAME);
+        } else {
+            sheet = workbook.getSheet(REQUIRED_ROLE_MAPPING_SHEET_NAME);
+        }
         if (null == sheet) {
```

Comment:
Objects.IsNull you can use

Reply:
if we go inside the IsNull it is doing the same
`public static boolean isNull(Object obj) {
        return obj == null;
    }`
    any reason to use it ?

Reply:
it is using things aviable rather we writing.

Reply:
ok then.. I shall use Objects.IsNull